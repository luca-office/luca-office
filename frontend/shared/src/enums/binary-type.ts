export enum BinaryType {
  Image = "Image",
  Video = "Video",
  PDF = "PDF"
}
