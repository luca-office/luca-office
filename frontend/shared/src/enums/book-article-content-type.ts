export enum BookArticleContentType {
  Text,
  Image,
  Video,
  Pdf
}
