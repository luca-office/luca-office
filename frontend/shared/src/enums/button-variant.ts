export enum ButtonVariant {
  Primary,
  Secondary,
  Ghost,
  IconOnly
}
