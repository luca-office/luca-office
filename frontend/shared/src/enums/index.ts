export * from "./binary-type"
export * from "./book-article-content-type"
export * from "./button-variant"
export * from "./email-message-template"
export * from "./erp-entry-id"
export * from "./erp-navigation-entry-type"
export * from "./erp-type"
export * from "./file-type"
export * from "./heading-level"
export * from "./icon-name"
export * from "./icon-position"
export * from "./input-type"
export * from "./intervention-group-type"
export * from "./node-type"
export * from "./notification"
export * from "./office-tool"
export * from "./progress-indicator-status"
export * from "./project-progress"
export * from "./rater-mode"
export * from "./rating-action-option"
export * from "./rating-status"
export * from "./reference-book-content-type"
export * from "./scenario-state"
export * from "./sorting"
export * from "./time-unit"
export * from "./upload-file"
export * from "./viewer-tools-type"
