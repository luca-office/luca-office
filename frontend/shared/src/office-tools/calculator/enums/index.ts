export * from "./calculator-button-type"
export * from "./calculator-key"
