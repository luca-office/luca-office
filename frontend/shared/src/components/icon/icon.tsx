import {css} from "@emotion/react"
import {kebabCase} from "lodash-es"
import * as React from "react"
import {IconName} from "../../enums"
import {CustomStyle, iconDefaultColor, spacingTiny} from "../../styles"

interface IconProps {
  readonly className?: string
  readonly color?: string
  readonly hoverColor?: string
  readonly customSpacing?: number
  readonly fillColor?: string
  readonly hasSpacing?: boolean
  readonly height?: number
  readonly onClick?: (event: React.MouseEvent<HTMLDivElement>) => void
  readonly title?: string
  readonly width?: number
}

interface Props extends IconProps, CustomStyle {
  readonly name: IconName
}

export const Icon: React.FC<Props> = props => (
  <div
    onClick={props.onClick}
    title={props.title}
    className={props.className}
    css={[iconStyle(props), props.customStyles]}
    dangerouslySetInnerHTML={{__html: content(props.name)}}
  />
)

// Dynamic import to avoid huge switch case here
const content = (iconName: IconName) => {
  const name = kebabCase(iconName)
  return require(`../../assets/images/icons/${name}.svg`)
}

Icon.defaultProps = {
  width: 16,
  height: 16
}

const iconStyle = (props: Props) =>
  css({
    height: props.height,
    width: props.width,
    minWidth: props.width,
    color: props.color ?? iconDefaultColor,
    ...(props.hasSpacing ? {marginRight: props.customSpacing || spacingTiny} : {}),

    ...(!!props.hoverColor && {
      "&:hover": {
        color: props.hoverColor
      }
    })
  })
