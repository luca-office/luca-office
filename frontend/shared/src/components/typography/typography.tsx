import {css} from "@emotion/react"
import * as React from "react"
import {DetailedHTMLProps, HTMLAttributes} from "react"
import {HeadingLevel} from "../../enums"
import {Children, CustomStyle, fontColor, fontFamily, FontWeight, primaryColor, TextSize} from "../../styles"

interface HeadingProps extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
  readonly fontWeight?: FontWeight
  readonly level: HeadingLevel
  readonly color?: "primary"
  readonly className?: string
}

interface TextProps {
  readonly size?: TextSize
  readonly className?: string
  readonly title?: string
  readonly children: TextChildrenProps
}

export type TextChildrenProps = React.ReactElement | React.ReactElement[] | string | number | null | React.ReactNode

type CombinedTextProps = Pick<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, "onClick"> &
  CustomStyle &
  TextProps

export const Text: React.FC<CombinedTextProps> = ({
  children,
  onClick,
  customStyles,
  size = TextSize.Smaller,
  className,
  title
}) => (
  <div className={className} css={[styles.text, {fontSize: size}, customStyles]} title={title} onClick={onClick}>
    {children}
  </div>
)

export const Heading: React.FC<HeadingProps & CustomStyle> = props => (
  <div css={[styles.heading(props), props.customStyles]} onClick={props.onClick} className={props.className}>
    {props.children}
  </div>
)

const getHeadingFontSize = (headingLevel: HeadingLevel) => {
  switch (headingLevel) {
    case HeadingLevel.h1:
      return TextSize.Larger
    case HeadingLevel.h2:
      return TextSize.Large
    case HeadingLevel.h3:
      return TextSize.Medium
    case HeadingLevel.h4:
      return TextSize.Small
    case HeadingLevel.h5:
      return TextSize.Smaller
  }
}

const styles = {
  text: css({
    color: fontColor,
    fontWeight: "normal",
    fontFamily: fontFamily,
    overflowWrap: "break-word"
  }),
  heading: (props: HeadingProps) =>
    css({
      fontWeight: props.fontWeight ? props.fontWeight : FontWeight.Regular,
      color: props.color ? primaryColor : fontColor,
      fontSize: getHeadingFontSize(props.level),
      fontFamily: fontFamily,
      userSelect: "none"
    })
}
