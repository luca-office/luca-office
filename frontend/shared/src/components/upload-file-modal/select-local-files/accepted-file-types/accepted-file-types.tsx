import {css} from "@emotion/react"
import * as React from "react"
import {HeadingLevel, UploadFileType as FileType} from "../../../../enums"
import {
  CustomStyle,
  Flex,
  fontColorLight,
  shadowedCard,
  spacingCard,
  spacingHuge,
  spacingMedium,
  spacingSmall,
  spacingTiny,
  textEllipsis,
  TextSize
} from "../../../../styles"
import {useLucaTranslation} from "../../../../translations"
import {getAcceptedFileExtensions, getIconNameFromFileType, getLanguageKeyFromFileType} from "../../../../utils"
import {Icon} from "../../../icon/icon"
import {Heading, Text} from "../../../typography/typography"

interface Props extends CustomStyle {
  readonly fileTypes: FileType[]
  readonly isLimitedToSingleItem?: boolean
}

export const AcceptedFileTypes: React.FC<Props> = ({fileTypes, isLimitedToSingleItem = false, customStyles}) => {
  const {t} = useLucaTranslation()

  return (
    <div css={customStyles}>
      <Text customStyles={{marginBottom: spacingTiny}} size={TextSize.Small}>
        {isLimitedToSingleItem
          ? t("files_and_directories__upload_modal_file_typ_text_singular")
          : t("files_and_directories__upload_modal_file_typ_text_plural")}
      </Text>
      <div css={[shadowedCard, styles.wrapper]}>
        {fileTypes.map(fileType => (
          <div key={fileType} css={[Flex.row, styles.fileType]}>
            <Icon customStyles={[styles.icon]} name={getIconNameFromFileType(fileType)} />
            <Heading customStyles={styles.title} level={HeadingLevel.h3}>
              {t(getLanguageKeyFromFileType(fileType))}
            </Heading>
            <Heading customStyles={styles.subTitle} level={HeadingLevel.h3}>
              {/* TODO LUCA-2757 fix mime type */}
              {`(${Object.values(getAcceptedFileExtensions(fileType)).join(", ")})`}
            </Heading>
          </div>
        ))}
      </div>
    </div>
  )
}

const styles = {
  wrapper: css({
    display: "flex",
    justifyContent: "flex-start",
    alignItems: "center",
    padding: spacingMedium,
    paddingTop: spacingSmall,
    flexWrap: "wrap"
  }),
  fileType: css({justifyContent: "center", marginRight: spacingHuge, marginTop: spacingSmall}),
  icon: css({marginRight: spacingTiny}),
  text: css({
    padding: spacingCard,
    paddingTop: 0
  }),
  title: css(textEllipsis),
  subTitle: css(textEllipsis, {
    color: fontColorLight,
    marginLeft: spacingSmall
  })
}
