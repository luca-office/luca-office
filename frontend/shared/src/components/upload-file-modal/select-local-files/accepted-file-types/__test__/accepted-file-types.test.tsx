import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {UploadFileType as FileType} from "../../../../../enums"
import {Icon} from "../../../../icon/icon"
import {Heading, Text} from "../../../../typography/typography"
import {AcceptedFileTypes} from "../accepted-file-types"

const component = <AcceptedFileTypes fileTypes={[FileType.Video]} />

describe("accepted-file-types", () => {
  it("renders correctly", () => {
    const tree = create(component)
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", () => {
    const tree = shallow(component)
    expect(tree.find(Text)).toHaveLength(1)
    expect(tree.find(Icon)).toHaveLength(1)
    expect(tree.find(Heading)).toHaveLength(2)
  })
})
