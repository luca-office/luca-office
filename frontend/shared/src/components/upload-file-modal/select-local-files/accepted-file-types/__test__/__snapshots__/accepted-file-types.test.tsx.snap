// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepted-file-types renders correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-1i8w75j-Text"
  >
    files_and_directories__upload_modal_file_typ_text_plural
  </div>
  <div
    className="css-1abrcxf-AcceptedFileTypes"
  >
    <div
      className="css-170cyps-AcceptedFileTypes"
    >
      <div
        className="css-ijll0b-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1i6jmor-Heading"
      >
        files_and_directories__file_type_video
      </div>
      <div
        className="css-oxkgat-Heading"
      >
        (.mp4)
      </div>
    </div>
  </div>
</div>
`;
